### PR TITLE
Include inline auth for git: and https: if present.

### PR DESCRIPTION
**Pull request by othiym23**
Sets it in `git:` and `https:` URLs, as well as GitHub raw page paths (although I'm not sure that's something GitHub even supports anymore).

**r**: @iarna

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -31,10 +32,14 @@ exports.fromUrl = function (giturl) {
   var parsed = parseGitUrl(maybeGitHubShorthand(giturl) ? 'github:' + giturl : giturl)
   var matches = Object.keys(gitHosts).map(function (V) {
     var gitHost = gitHosts[V]
+    var auth = null
+    if (parsed.auth && (parsed.protocol === 'git:' || parsed.protocol === 'https:')) {
```
Comment (iarna) [inline index.js]: this should probably include http too…

**Comment by iarna** [comment]
LGTM, I'll merge this shortly.

**Comment by iarna** [comment]
Landed as 504cd4e1f2fef80ee48ed95e5213a227138af7d6

File: index.js (hunk 1 above)
Comment (othiym23) [inline index.js]: Good catch! Thank you for fixing this up!

**Pull request closed without merging**